docs/examples/201/web-app-conditional-log/main.bicep

```diff
@@ -0,0 +1,109 @@
+param skuName string = 'S1'
+param skuCapacity int = 1
+param location string = resourceGroup().location
+param appName string = uniqueString(resourceGroup().id)
+param logging bool = false
+
+var appServicePlanName = toLower('asp-${appName}')
+var webSiteName = toLower('wapp-${appName}')
+var appInsightName = toLower('appi-${appName}')
+var logAnalyticsName = toLower('la-${appName}')
+
+resource appServicePlan 'Microsoft.Web/serverfarms@2020-06-01' = {
+  name: appServicePlanName // Globally unique storage account name
+  location: location // Azure Region
+  sku: {
+    name: skuName
+    capacity: skuCapacity
+  }
+  tags: {
+    displayName: 'HostingPlan'
+    ProjectName: appName
+  }
+}
+
+resource appService 'Microsoft.Web/sites@2020-06-01' = {
+  name: webSiteName
+  location: location
+  identity: {
+    type: 'SystemAssigned'
+  }
+  tags: {
+    displayName: 'Website'
+    ProjectName: appName
+  }
+  properties: {
+    serverFarmId: appServicePlan.id
+    httpsOnly: true
+    siteConfig: {
+      minTlsVersion: '1.2'
+    }
+  }
+}
+resource appServiceLogging 'Microsoft.Web/sites/config@2020-06-01' = {
+  name: '${appService.name}/logs'
+  properties: {
+    applicationLogs: {
+      fileSystem: {
+        level: 'Warning'
+      }
+    }
+    httpLogs: {
+      fileSystem: {
+        retentionInMb: 40
+        enabled: true
+      }
+    }
+    failedRequestsTracing: {
+      enabled: true
+    }
+    detailedErrorMessages: {
+      enabled: true
+    }
+  }
+}
+
+resource appServiceAppSettings 'Microsoft.Web/sites/config@2020-06-01' = if (logging) {

```

Comment:
since the `if (logging)` condition is used multiple times, it may make sense to put all of those resources in a module and then conditionally deploy the module. That is a personal preference, so I will leave that to you

Reply:
That makes sense and can see that viewpoint.  I think part of the question is overall how modules should be scoped.  So far I've taken it from the individual resources as opposed to in this case the feature set of logging.  Think realistically for this one a single logging module makes more sense.

Thanks for the feedback! I will update accordingly.